libraries/typescript/Microsoft.Bot.Builder.Adapters.WeChat/src/accessTokenStorage.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ * @module wechat
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Storage, StoreItems } from 'botbuilder-core';
+import { WeChatAccessToken } from './weChatSchema';
+
+/**
+ * Storage provider for access token.
+ */
+export class AccessTokenStorage {
+    private Storage: Storage;
+
+    /**
+     * Creates an instance of access token storage.
+     * @param storage
+     */
+    constructor(storage: Storage) {
+        this.Storage = storage;
+    }
+
+    /**
+     * Saves store items to storage.
+     * @param key Item key to write to the storage.
+     * @param value Item value to write to the storage.
+     */
+    public async SaveAsync(key: string, value: WeChatAccessToken): Promise<void> {
+        const dict: StoreItems = {
+            [key]: value
+        };
+        await this.Storage.write(dict);
+    }
+
+    /**
+     * Loads store items from storage.
+     * @param key Item key to read from the store.
+     */
+    public async GetAsync(key: string): Promise<WeChatAccessToken> {
+        const result: StoreItems = await this.Storage.read([key]);
+        if (result[key] === undefined) {
+            return result[key];
+        } else {
+            const weChatResult = new WeChatAccessToken(result[key]);
```

Comment:
>new WeChatAccessToken(result[key]) [](start = 33, length = 34)

We don't typically use classes for data in JavaScript. That's a C# pattern. This really should just be an interface and not a class. The whole implementation should be this simple:

public async get(key: string): Promise<WeChatAccessToken|undefined> {
     const result: StoreItems = await this.storage.read([key]);
     return result[key] as WeChatAccessToken;
}
 
You're basically just casting the object that was read in from storage to an instance of WeChatAccessToken. The actual return type adds a union of undefined which says that the method may return undefined and callers should check for undefined before using.